gobblin-metastore/src/main/java/gobblin/metastore/util/StateStoreCleaner.java

```diff
@@ -82,48 +90,79 @@ public StateStoreCleaner(Properties properties) throws IOException {
     this.retentionTimeUnit = TimeUnit.valueOf(properties.getProperty(STATE_STORE_CLEANER_RETENTION_TIMEUNIT_KEY,
         DEFAULT_STATE_STORE_CLEANER_RETENTION_TIMEUNIT).toUpperCase());
 
+    this.cleanerIntervalInMinutes = Long.parseLong(properties.getProperty(
+            STATE_STORE_CLEANER_INTERVAL_IN_MINUTES_KEY, DEFAULT_STATE_STORE_CLEANER_INTERVAL_IN_MINUTES));
+    this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor(
+        ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("StateStoreCleanerScheduler")));
+
     this.cleanerRunnerExecutor = Executors.newFixedThreadPool(
         Integer.parseInt(properties.getProperty(
             STATE_STORE_CLEANER_EXECUTOR_THREADS_KEY, DEFAULT_STATE_STORE_CLEANER_EXECUTOR_THREADS)),
         ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("StateStoreCleaner")));
 
     URI fsUri = URI.create(properties.getProperty(
         ConfigurationKeys.STATE_STORE_FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI));
-    this.fs = FileSystem.get(fsUri, new Configuration());
+    this.fs = FileSystem.get(fsUri, HadoopUtils.getConfFromProperties(properties));
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    log.info("Starting the " + StateStoreCleaner.class.getSimpleName());
```

Comment:
This was not intended, but a side effect of trying to handle 8 different year old PRs at once.